pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+\end{definition}
+
+\begin{definition}[\texttt{f32}]$\sfi$
+    \texttt{f32} is the Rust 32-bit floating point type. \inRust{f32}{https://doc.rust-lang.org/std/primitive.f32.html}
+\end{definition}
+
+\begin{definition}[\texttt{f64}]$\sfi$
+    \texttt{f64} is the Rust 64-bit floating point type. \inRust{f64}{https://doc.rust-lang.org/std/primitive.f64.html} 
+\end{definition}
+
+\todonei{Add / pointers to  ``binary64” type defined in IEEE 754-2008.}
+
+\begin{definition}[\texttt{IntDistance}]
+\texttt{IntDistance} is equivalent to \texttt{u32}.
+\end{definition}
+
+\begin{definition}[\texttt{null}]
+    Represents a missing or unknown value. It is not equivalent to any other value, and is counted as an element when calculating symmetric distance.
+\end{definition}
+
+\begin{definition}[\texttt{u32}]$\sfi$
+    u32 is the Rust 32-bit unsigned integer type. If $v$ is a value of type \texttt{u32}, then we know that $v\in \{0,1,2,\ldots, 2^{32}-1\}$. \inRust{u32}{https://doc.rust-lang.org/std/primitive.u32.html}
+\end{definition}
+
+\begin{definition}[\texttt{usize}]
+    \texttt{usize} is defined differently on 32-bit and 64-bit machines. This is because the size of this primitive is equal to the number of bytes it takes to reference any location in memory.
+    \begin{itemize}
+        \item \textbf{32-bit machines:} if \texttt{v} is a value of type \texttt{usize}, then we know that $\texttt{v}\in \{0,1,2,\ldots,2^{32} - 1\}$
+        \item \textbf{64-bit machines:} if \texttt{v} is a value of type \texttt{usize}, then we know that $\texttt{v}\in \{0,1,2,\ldots,2^{64} - 1\}$
+    \end{itemize}
+    \inRust{usize}{https://doc.rust-lang.org/std/primitive.usize.html}
+\end{definition}
+
+\begin{definition}[\texttt{Vec(T)}]$\sfi$
+    The rust type \texttt{Vec(T)} consists of ordered lists of type \texttt{T}. For example, if \texttt{T = bool}, then values of type \texttt{Vec(T)} include $\texttt{[], [0], [1], [0, 0],} \ldots$ \inRust{Vec<T>}{https://doc.rust-lang.org/std/vec/struct.Vec.html}
+\end{definition}
+
+\horizline
+
+\subsection{Notes, todos, questions}
+
+%\silvia{Discuss: bounded vs lower bound and upper bound}
+
+\todonei{Include info on MPFR, and possibly relate it to our existing definitions of floats.}
+
+\todonei{Define plus, minus, etc. below each type on which they operate. For example, the definition for \texttt{u32} should also include a definition of plus on \texttt{u32}, multiplication on \texttt{u32}, etc.}
+
+\questionr{Should we have a general definition for ``floats'' (and ``integers''?), or is it sufficiently understood what a float is in general?}
+
+\section{Domains}
+
+$\sfi$ A \emph{data domain} is a representation of the set of values on which a metric or function can operate. For example, if a function accepts inputs from the domain \texttt{IntervalDomain(1:u32,17:u32)}, this means that the function can take any input value \texttt{v} of type \texttt{u32} such that \texttt{1 <= v} and \texttt{v <= 17}.
```

Comment:
"metric" -> "transformation"

Reply:
I think metric is correct here, because when `SymmetricDistance` is the `input_metric`, we talk about its `input_domain`. I also think that the term "function" should capture transformations since transformations are implemented as functions.

Feel free to reply with additional comments, and let me know if we should, for example, change "metric" to "transformation".

Reply:
I see.  How about "the set of values on which the function associated with a transformation or measurement can operate.  Each metric (cross-reference section) is associated to certain data domains."  (What was confusing me is that metrics don't "operate" in the framework.)

Reply:
I understand -- that makes sense. The new explanation of domains looks great!